eqs/src/eth_polling.rs

```diff
@@ -183,6 +212,16 @@ impl EthPolling {
                         })
                         .collect();
 
+                    let mut updated_state = self.query_result_state.write().await;
```

Comment:
I'm not a fan of this change. In the old code, the memos were being created before the QueryResultState saw any changes, which means eqs could continue serving queries. We could read the block_id in the same operation that we use to build the MerkleTree above, and avoid the early write lock. This has the added advantage of not changing the state at all if something goes wrong, but that's only semi-useful, since we don't really have any recoverable errors in this code segment. In fact, I'd actually prefer to see the Commit event created on the stack before the write() lock as well...

Reply:
Yeah, fair point. I'll do it that way